plugins/vpp/ipfixplugin/descriptor/flowprobe_feature.go

```diff
@@ -0,0 +1,108 @@
+)
+
+// Validation errors:
+var (
+	// ErrIfaceNotDefined returned when interface in confiugration is empty string.
+	ErrIfaceNotDefined = errors.New("missing interface name for Flowprobe Feature")
+)
+
+// FPFeatureDescriptor configures Flowprobe Feature for VPP.
+type FPFeatureDescriptor struct {
+	ipfixHandler vppcalls.IpfixVppAPI
+	log          logging.Logger
+}
+
+// NewFPFeatureDescriptor creates a new instance of FPFeatureDescriptor.
+func NewFPFeatureDescriptor(ipfixHandler vppcalls.IpfixVppAPI, log logging.PluginLogger) *kvs.KVDescriptor {
+	ctx := &FPFeatureDescriptor{
+		ipfixHandler: ipfixHandler,
+		log:          log.NewLogger("flowprobe-feature-descriptor"),
+	}
+	typedDescr := &adapter.FlowProbeFeatureDescriptor{
+		Name:          FPFeatureDescriptorName,
+		NBKeyPrefix:   ipfix.ModelFlowprobeFeature.KeyPrefix(),
+		ValueTypeName: ipfix.ModelFlowprobeFeature.ProtoName(),
+		KeySelector:   ipfix.ModelFlowprobeFeature.IsKeyValid,
+		KeyLabel:      ipfix.ModelFlowprobeFeature.StripKeyPrefix,
+		WithMetadata:  true,
+		Validate:      ctx.Validate,
+		Create:        ctx.Create,
+		Delete:        ctx.Delete,
+		Retrieve:      ctx.Retrieve,
+		Dependencies:  ctx.Dependencies,
+	}
+	return adapter.NewFlowProbeFeatureDescriptor(typedDescr)
+}
+
+// Validate checks if Flowprobe Feature configuration is good to send to VPP.
+func (d *FPFeatureDescriptor) Validate(key string, value *ipfix.FlowProbeFeature) error {
+	if value.GetInterface() == "" {
+		return kvs.NewInvalidValueError(ErrIfaceNotDefined, "interface")
+	}
+	return nil
+}
+
+// Create uses vppcalls to pass Flowprobe Feature configuration for interface to VPP.
+func (d *FPFeatureDescriptor) Create(key string, val *ipfix.FlowProbeFeature) (metadata interface{}, err error) {
+	err = d.ipfixHandler.AddFPFeature(val)
+	return val, err
+}
+
+// Delete uses vppcalls to remove Flowprobe Feature configuration for interface.
+func (d *FPFeatureDescriptor) Delete(key string, val *ipfix.FlowProbeFeature, metadata interface{}) (err error) {
+	err = d.ipfixHandler.DelFPFeature(val)
+	return
+}
+
+// Dependencies sets Flowprobe Params as a dependency which must be created
+// before enabling Flowprobe Feature on an interface.
+func (d *FPFeatureDescriptor) Dependencies(key string, val *ipfix.FlowProbeFeature) []kvs.Dependency {
+	return []kvs.Dependency{
+		{
+			Label: "flowprobe-params",
+			Key:   ipfix.FlowprobeParamsKey(),
+		},
+	}
+}
+
+// Retrieve hopes that configuration in correlate is actual configuration in VPP.
+// As soon as VPP devs will add dump API calls, this methods should be fixed.
+// Also, this method sets metadata, so descriptor for Flowprobe Params would know
+// that there are some interfaces with Flowprobe Feature enabled.
+func (d *FPFeatureDescriptor) Retrieve(correlate []adapter.FlowProbeFeatureKVWithMetadata) (retrieved []adapter.FlowProbeFeatureKVWithMetadata, err error) {
+	for i := range correlate {
+		correlate[i].Metadata = correlate[i].Value
```

Comment:
Please don't change the input argument inside a descriptor (even though it doesn't seem to cause an issue in this case).
Build new slice instead, i.e.:
```
for i := range correlate {
    retrieved = append(retrieved,  adapter.FlowProbeFeatureKVWithMetadata{
        Value: correlate[i].Value,
        // etc.
    })
}
```